RoyalRoad.java

```diff
@@ -7,60 +7,94 @@
 import java.util.*;
 import java.text.*;
 
-public class OneMax extends FitnessFunction{
-
+public class RoyalRoad extends FitnessFunction {
 /*******************************************************************************
 *                            INSTANCE VARIABLES                                *
 *******************************************************************************/
-
-
-/*******************************************************************************
-*                            STATIC VARIABLES                                  *
-*******************************************************************************/
-
+    private List<String> patterns;
+    private int reward;
+    private String variant;
+    private int buildingBlockSize = 8;
 
 /*******************************************************************************
 *                              CONSTRUCTORS                                    *
 *******************************************************************************/
-
-	public OneMax(){
-		name = "OneMax Problem";
-	}
+    public RoyalRoad(String variant) {
+        this.name = "Royal Road";
+        this.variant = variant;
+        initializePatternsAndRewards();
+    }
 
 /*******************************************************************************
 *                                MEMBER METHODS                                *
 *******************************************************************************/
-
-//  COMPUTE A CHROMOSOME'S RAW FITNESS *************************************
-
-	public void doRawFitness(Chromo X){
-
-		X.rawFitness = 0;
-		for (int z=0; z<Parameters.numGenes * Parameters.geneSize; z++){
-			if (X.chromo.charAt(z) == '1') X.rawFitness += 1;
-		}
-	}
-
-//  PRINT OUT AN INDIVIDUAL GENE TO THE SUMMARY FILE *********************************
-
-	public void doPrintGenes(Chromo X, FileWriter output) throws java.io.IOException{
+    private void initializePatternsAndRewards() {
```

Comment:
I think we can revert this to the original OneMax functions and just change how fitness is calculated to count the number of blocks of 1's instead of each 1 individually. I will have to write out what I'm thinking and send it in a separate commit. Also, I don't understand the "11110000" part of the RR2 functions.